service/src/main/java/org/cbioportal/service/impl/CoExpressionServiceImpl.java

```diff
@@ -160,29 +173,52 @@
             selectedSampleIdsMap.put(samples.get(lc).getInternalId(), lc);
         }
 
+        // These next few lines build a list of indices into the genetic_alteration.VALUES
+        // column by iterating over all the samples in the molecular profile (method parameter)
+        // and selecting only samples that are included in the user query.
         Set<Integer> includedIndexes = new HashSet<>();
         for (Integer internalSampleId : internalSampleIds) {
             if (selectedSampleIdsMap.containsKey(internalSampleId)) {
                 includedIndexes.add(internalSampleIdToIndexMap.get(internalSampleId));
             }
         }
-        
+
         Boolean isMolecularProfileBOfGenesetType = molecularProfile.getMolecularAlterationType()
-                .equals(MolecularProfile.MolecularAlterationType.GENESET_SCORE);
+            .equals(MolecularProfile.MolecularAlterationType.GENESET_SCORE);
+
+
+        // These next few lines filter out genetic_alteration values from the query gene/geneset
+        // genetic_alteration.VALUES column by considering only the indices of the samples in the user query.
         List<String> queryValues = Arrays.asList(queryMolecularDataList.getSplitValues());
         List<String> includedQueryValues = includedIndexes.stream().map(index -> queryValues.get(index))
-                .collect(Collectors.toList());
+            .collect(Collectors.toList());
 
-        Map<String,List<String>> values = new HashMap<String,List<String>>();
-        for (String entityId : molecularDataMap.keySet()) {
-            List<String> internalValues = new ArrayList<>(
-                    Arrays.asList(molecularDataMap.get(entityId).getSplitValues()));
-            List<String> includedInternalValues = includedIndexes.stream().map(index -> internalValues.get(index))
-                    .collect(Collectors.toList());
-            values.put(entityId, includedInternalValues);
+        // Get an iterator to all the MolecularAlteration (genetic_alteration table records) in the profile
+        if (geneticEntityType.equals(GeneticEntityType.GENE)) {
+            maItr = molecularDataService.getMolecularAlterations(molecularProfileId, null, "SUMMARY");
+        } else if (geneticEntityType.equals(GeneticEntityType.GENESET)) {
+            maItr = genesetDataService.getGenesetAlterations(molecularProfileId, null);
         }
-        coExpressionList = computeCoExpressions(values, includedQueryValues, isMolecularProfileBOfGenesetType, threshold);  
-        return coExpressionList;
+
+        // For each MolecularAlteration in the profile, compute a CoExpression to return.
+        // If the MolecularAlteration is for the query gene/geneset, skip it.  Otherwise,
+        // filter out genetic_alteration values from genetic_alteration values from
```

Comment:
accidentally repeated some words in this comment

Reply:
Thanks for catching, just updated.